src/App/component/AddRepositoryDialog.jsx

```diff
@@ -7,63 +7,261 @@ import {
-        repositoryURL: repositoryURL
+
+      if (githubRepositoryURL !== "") {
+        checker.push(checkGithubRepositoryURL());
+      }
+      if (gitlabRepositoryURL !== "") {
+        checker.push(checkGitlabRepositoryURL());
       }
-      Axios.post(`http://localhost:9100/pvs-api/project/${projectId}/repository/${repoType}`, payload,
-        {headers: {"Authorization": `${jwtToken}`}})
-        .then(() => {
-          reloadProjects()
-          handleClose()
-        })
-        .catch((e) => {
-          alert(e.response.status)
-          console.error(e)
-        })
+      if (sonarRepositoryURL !== "") {
+        checker.push(checkSonarRepositoryURL());
+      }
+
+      Promise.all(checker)
+        .then((response) => {
+          if (response.includes(false) === false) {
+            let payload = {
+              projectId: projectId,
+              repositoryURL: repositoryURL
+            }
+
+            Axios.post(`http://localhost:9100/pvs-api/project/${projectId}/repository/${repoType}`, payload,
+              {headers: {"Authorization": `${jwtToken}`}})
+              .then(() => {
+                reloadProjects()
+                handleClose()
+              })
+              .catch((e) => {
+                alert(e.response.status)
+                console.error(e)
+              })
+          }
+        }).catch((e) => {
+        alert(e.response.status)
+        console.error(e)
+      })
     }
   }
 
+  const checkGithubRepositoryURL = () => {
+    return Axios.get(`http://localhost:9100/pvs-api/repository/github/check?url=${githubRepositoryURL}`,
+      {headers: {"Authorization": `${jwtToken}`}})
+      .then(() => {
+        setIsGithubAvailable(true);
+        return true
+      })
+      .catch(() => {
+        alert("github error")
+        return false
+      })
+  }
+
+  const checkGitlabRepositoryURL = () => {
+    return Axios.get(`http://localhost:9100/pvs-api/repository/gitlab/check?url=${gitlabRepositoryURL}`,
+      {headers: {"Authorization": `${jwtToken}`}})
+      .then(() => {
+        setIsGitlabAvailable(true);
+        return true
+      })
+      .catch(() => {
+        alert("gitlab error")
+        return false
+      })
+  }
+
+  const checkSonarRepositoryURL = () => {
+    return Axios.get(`http://localhost:9100/pvs-api/repository/sonar/check?url=${sonarRepositoryURL}`,
+      {headers: {"Authorization": `${jwtToken}`}})
+      .then(() => {
+        setIsSonarAvailable(true);
+        return true
+      })
+      .catch((e) => {
+        alert("sonar error")
+        console.error(e)
+        return false
+      })
+  }
+
+  const onClick1 = () => {
+    setShowGithubDiv(true)
+    setShowGitlabDiv(false)
+    setShowSonarDiv(false)
+  }
+  const onClick2 = () => {
+    setShowSonarDiv(true)
+    setShowGithubDiv(false)
+    setShowGitlabDiv(false)
+  }
+  const onClick3 = () => {
+    setShowGitlabDiv(true)
+    setShowGithubDiv(false)
+    setShowSonarDiv(false)
+  }
```

Comment:
Please help rename those three methods to a more clear action-result name instead of the currently number-oriented namings.